CanaryValidator/Canary.Tests.ps1

```diff
@@ -390,7 +390,85 @@ while ($runCount -le $NumberOfIterations)
                 $asTenantSubscription | Select-AzureRmSubscription -ErrorAction Stop
             }           
         } 
+	
+	Invoke-Usecase -Name 'RoleAssignmentAndCustomRoleDefinition' -Description "Assign a reader role and create a custom role definition" -UsecaseBlock `
+        {
+            $readerRoleName = (Get-AzureRmRoleDefinition -Name Reader).Name
+            $subscriptionID = (Get-AzureRmSubscription -SubscriptionName $tenantSubscriptionName).SubscriptionId
+            $servicePrincipal = (Get-AzureRmADServicePrincipal)[0]
+            $customRoleName = "CustomCanaryRole-" + [Random]::new().Next(1,99)
```

Comment:
The problem of using random value is that it cannot be easily found and cleaned up if your test does not finish. Can't you instead use a constant name and clean it up at the end (and check for its existence in the beginning in case it was not cleaned up by previous run?